docs/proposals/allocation-strategies.md

```diff
@@ -0,0 +1,106 @@
+# Allocation strategies
+
+<!-- toc -->
+
+<!-- /toc -->
+
+## Summary
+
+Allocation strategies is a feature to allow choosing how to allocate LVs from the VG when using TopoLVM.
+For example, keep the LV on a single PV or spread it over multiple.
+
+## Motivation
+
+Some applications benefit from "owning" the disk they use and keeping the data on a single physical disk.
+In other situations it may be better spread the data for fault tolerance.
+
+### Goals
+
+- Make it possible to reserve a whole PV for one LV (`dedicated`).
+- Expose the `--type` flag of `lvcreate` to the TopoLVM user.
```

Comment:
Have you looked at the `--alloc` flag for `lvcreate` ? In particular, `--alloc contiguous` or `--alloc cling` means that the physical extents for the logical volume will be on same physical volume. Would this be sufficient to meet the requirement in the `Motivation`? If it's enough, maybe you wouldn't have to change any scheduling logic after all.

Reply:
I did but I don't think it is quite enough for "full disk allocation". If the only requirement would be to make sure that the requested LV uses a single PV I think it would be enough maybe, but if it must also be the *only* LV using that PV then we still need something more. And I believe this was a requirement.

Reply:
Now that I have thought about it a bit more, I think it would make sense to expose `--alloc` instead of `--type` at this point. It is more "in tune" with the full disk allocation feature and would allow fine tuning for better allocation efficiency when it is ok to share one PV among multiple LVs.